lib/src/Base/Optim/Bonmin.cxx

```diff
@@ -0,0 +1,312 @@
+        throw InvalidArgumentException(HERE) << "Unknown solver " << algoName;
+      }
+    }
+      
+    algoName_ = algoName;
+  }
+  
+  String Bonmin::getAlgorithmName() const
+  {
+    return algoName_;
+  }
+
+  
+/** Check whether this problem can be solved by this solver. */
+void Bonmin::checkProblem(const OptimizationProblem & problem) const
+{
+  // Cannot solve multi-objective problems
+  if (problem.hasMultipleObjective()) throw InvalidArgumentException(HERE) << "Error: Bonmin algorithm " << algoName_ << " does not support multi-objective optimization";
+  
+  // No LeastSquaresProblem / NearestPointProblem
+  if (problem.hasResidualFunction() || problem.hasLevelFunction())
+    throw InvalidArgumentException(HERE) << "Bonmin does not support least squares or nearest point problems";
+}
+
+
+#ifdef OPENTURNS_HAVE_BONMIN
+
+/** Accessors to Bonmin options */
+void getOptionsFromResourceMap(BonminSetup & bonmin)
+{
+//   Get options for Bonmin setup from ResourceMap
+//   See Bonmin/Ipopt user manuals for more details.
+  
+  std::vector<String> keys(ResourceMap::GetKeys());
+  UnsignedInteger nbKeys = keys.size();
+    
+  for (UnsignedInteger i=0; i<nbKeys; ++i)
+    if (keys[i].substr(0, 7) == "Bonmin-")
+    {
+      String optionName(keys[i].substr(7));
+      String type(ResourceMap::GetType(keys[i]));
+      if (type == "string")
+        bonmin.options()->SetStringValue(optionName, ResourceMap::GetAsString(keys[i]));
+      else if (type == "float")
+        bonmin.options()->SetNumericValue(optionName, ResourceMap::GetAsScalar(keys[i]));
+      else if (type == "unsigned int")
+        bonmin.options()->SetIntegerValue(optionName, ResourceMap::GetAsUnsignedInteger(keys[i]));
+      else
+        throw InvalidArgumentException(HERE) << "Unsupported type " << type << " for Bonmin option " << optionName;
+    }
+}
+
+#endif
+
+/** Performs the actual computation. */
+void Bonmin::run()
+{
+  #ifdef OPENTURNS_HAVE_BONMIN
+  // Check problem
+  checkProblem(getProblem());
+  
+  // Check starting point
+  if (getStartingPoint().getDimension() != getProblem().getDimension())
+    throw InvalidArgumentException(HERE) << "Invalid starting point dimension (" << getStartingPoint().getDimension() << "), expected " << getProblem().getDimension();
+  
+  // Create BonminProblem
+  Ipopt::SmartPtr<BonminProblem> tminlp = new BonminProblem(getProblem(),getStartingPoint(), getMaximumEvaluationNumber());
+  tminlp->setProgressCallback(progressCallback_.first, progressCallback_.second);
+  tminlp->setStopCallback(stopCallback_.first, stopCallback_.second);
+                              
+  // Create setup, initialize options
+  BonminSetup bonmin;
+  bonmin.initializeOptionsAndJournalist();
+   
+  // Update algorithm name and other user defined options
+  bonmin.options()->SetStringValue("bonmin.algorithm", algoName_);
+  bonmin.options()->SetIntegerValue("max_iter", getMaximumIterationNumber());
+  
+  getOptionsFromResourceMap(bonmin);
```

Comment:
care here if `ResourceMap` contains `algoName` and/or `max_iter` keys

Reply:
Indeed.
I included a check for potential conflicts on these two specific parameters: if values from Bonmin attributes and ResourceMap don't match, a warning is raised and the value specified in the ResourceMap is ignored.